papers/hatra21/hatra21.tex

```diff
@@ -178,31 +194,26 @@ \subsection{Infallible types}
 where some subterms are \emph{flagged} as having type errors, written $\squnder{N}$. Write $\erase(N)$
 for the result of erasing flaggings: $\erase(\squnder{N}) = \erase(N)$.
 
-%% For example the usual
-%% type rules for field access becomes:
-%% \[
-%%   \infer{
-%%     \Gamma \vdash M \Rightarrow M' : T
-%%   }{
-%%     \Gamma \vdash M.\ell \Rightarrow M'.\ell : U
-%%   }
-%%   [
-%%     T = \{ \overline{\ell:U} \} \mbox{ and } (\ell:U) \in (\overline{\ell:U})
-%%   ]
-%% \]
-%% but there is also a rule for unsuccessful field access:  
-%% \[
-%%   \infer{
-%%     \Gamma \vdash M \Rightarrow M' : T
-%%   }{
-%%     \Gamma \vdash M.\ell \Rightarrow \squnder{M'.\ell} : U
-%%   }
-%%   [
-%%     T = \{ \overline{\ell:U} \} \mbox{ implies } \ell \not\in \overline{\ell}
-%%   ]
-%% \]
-%% In this type rule, $U$ is unconstrained.
-
+For example, in Lua, the $\STRING.\FIND$ function expects two strings, and returns the
+offsets for that string:
+\[
+  \STRING.\FIND(\strlit{hello}, \strlit{ell}) \rightarrow (2, 4)
+\]
+and in Luau it has the type:
+\[
+  \STRING.\FIND : (\STRING, \STRING) \rightarrow (\NUMBER?, \NUMBER?)
+\]
```

Comment:
Should we rewrite the code samples here as listings? If not, I think we can remove the `listing` package import.

Reply:
I prefer writing code in math mode, so I'd vote for removing the listings package.